nsqadmin/static/js/views/base.js

```diff
@@ -83,6 +85,29 @@ var BaseView = Backbone.View.extend({
         this.removed = true;
         this.removeSubviews();
         Backbone.View.prototype.remove.apply(this, arguments);
+    },
+
+    parseErrorMessage: function(jqXHR) {
+        var msg = 'ERROR: failed to connect to nsqadmin';
+        if (jqXHR.readyState === 4) {
+            try {
+                var parsed = JSON.parse(jqXHR.responseText);
+                msg = parsed['message'];
+            } catch(err) {
+                msg = 'ERROR: failed to decode JSON - ' + err.message;
+            }
+        }
+        return msg;
+    },
+
+    handleAJAXError: function(jqXHR) {
+        $('#warning, #error').hide();
+        $('#error .alert').text(this.parseErrorMessage(jqXHR));
+        $('#error').show();
+    },
+
+    handleViewError: function(jqXHR) {
+        this.render({'message': this.parseErrorMessage(jqXHR)});
```

Comment:
I'm seeing something like `this.template = require('./error.hbs')` appear before each time this method is called. Mutating `template` on these views puts them in a weird state. It might be a better practice to make `handleViewError` something like:
```
var errorTemplate = require('./error.hbs');

...

handleViewError: function(jqXHR) {
    this.removeSubviews();
    this.$el.html(errorTemplate({'message': this.parseErrorMessage(jqXHR)}));
}
```